### PR TITLE
ESS-983: As an end-user, when dataChannel for file transfer closes unexpected, it should trigger ERROR state

### PR DESCRIPTION
**Pull request by ishannaithani**
**Purpose of this PR**

When an active dataChannel between two peers is closed unexpectedly - the dataTransferState was not being trigger with a state of error `[Skylink.DATA_TRANSFER_STATE.ERROR]`. 

This PR adds the code that triggers the dataTransferState event with the required params in case the dataChannel is closed.

**Steps to reproduce**:
Reproduce and check with 1st Peer end:
1. Connect 2 peers to a room.
2. In the 1st peer, send a huge (of about 3GB) file.
3. Quickly, in the 1st peer's console, do the following: "skylink._dataChannels[2ndPeerID].main.channel.close()". Note that "skylink" here is an arbitrary value so please provide the Skylink object value itself.
4. Notice that "dataTransferState" does not trigger (state=ERROR) on the 1st peer or 2nd peer end.

Reproduce and check with 2nd Peer end:
1. Connect 2 peers to a room.
2. In the 1st peer, send a huge (of about 3GB) file.
3. Quickly, in the 2ns peer's console, do the following: "skylink._dataChannels[1stPeerID].main.channel.close()". Note that "skylink" here is an arbitrary value so please provide the Skylink object value itself.
4. Notice that "dataTransferState" does not trigger (state=ERROR) on the 1st peer or 2nd peer end.

**Expected result**:
It should trigger (state=ERROR) for "dataTransferState" event for both 1st/2nd peer end when it is closed abruptly on either 1st/2nd peer end.

**Observed result**:
It does not trigger the (state=ERROR) for "dataTransferState" event for either 1st/2nd peer end when it is closed abruptly on either 1st/2nd peer end.

### CODE REVIEW

File: source/data-transfer.js [hunk 1]
```diff
@@ -2280,11 +2280,20 @@ Skylink.prototype._handleDataTransferTimeoutForPeer = function (transferId, peer
...
+  var channelProp = channelType === self.DATA_CHANNEL_TYPE.MESSAGING ? 'main' : channelName;
+
+  // Safe access of _dataChannel object in case dataChannel has been closed unexpectedly | ESS-983
+  var objPeerDataChannel = self._dataChannels[peerId] || {};
+  if (objPeerDataChannel.hasOwnProperty(channelProp) && typeof objPeerDataChannel[channelProp] === 'object') {
+    transferId = objPeerDataChannel[channelProp].transferId;
+    streamId = objPeerDataChannel[channelProp].streamId;
+  }
+  else {
+    return; // dataChannel not avaialble propbably having being closed abruptly | ESS-983
```
Comment (geekavi) [inline source/data-transfer.js:2295]: please remove reference of ticket number

File: source/data-channel.js [hunk 2]
```diff
@@ -103,12 +103,26 @@ Skylink.prototype._createDataChannel = function(peerId, dataChannel, bufferThres
...
   var onCloseHandlerFn = function () {
-    log.debug([peerId, 'RTCDataChannel', channelProp, 'Datachannel has closed']);
+    var dcMessageStr = "Datachannel has closed";
+    var transferId = getTransferIDByPeerId(peerId);
+    log.debug([peerId, 'RTCDataChannel', channelProp, dcMessageStr]);
 
     self._trigger('dataChannelState', self.DATA_CHANNEL_STATE.CLOSED, peerId, null, channelName,
       channelType, null, self._getDataChannelBuffer(dataChannel));
 
+    // ESS-983 Handling dataChannel unexpected close to trigger dataTransferState Error.
```
Comment (geekavi) [inline source/data-channel.js:123]: please remove reference of ticket number

File: source/data-transfer.js [hunk 3]
```diff
@@ -2280,11 +2280,20 @@ Skylink.prototype._handleDataTransferTimeoutForPeer = function (transferId, peer
...
+  var streamId = null;
   var isStreamChunk = false;
+  var channelProp = channelType === self.DATA_CHANNEL_TYPE.MESSAGING ? 'main' : channelName;
+
+  // Safe access of _dataChannel object in case dataChannel has been closed unexpectedly | ESS-983
+  var objPeerDataChannel = self._dataChannels[peerId] || {};
+  if (objPeerDataChannel.hasOwnProperty(channelProp) && typeof objPeerDataChannel[channelProp] === 'object') {
+    transferId = objPeerDataChannel[channelProp].transferId;
+    streamId = objPeerDataChannel[channelProp].streamId;
+  }
```
Comment (geekavi) [inline source/data-transfer.js:2293]: please commit files only which you have worked, not auto generated files.

**Review by geekavi — CHANGES_REQUESTED** [review]

File: source/data-transfer.js (hunk 1 above)
Comment (ncurrier) [reply source/data-transfer.js:2295]: ticket number is ok

File: source/data-transfer.js (hunk 3 above)
Comment (ishannaithani) [reply source/data-transfer.js:2293]: @geekavi I've reverted the auto generated files and pushed them. Please check now if this is okay?

**Comment by geekavi** [comment]
looks good!

**Pull request merged**